src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,102 @@
+package storage; /**
+ * This class handles any files related process for Snom
+ * Eg. Create, Write, Delete save files.
+ *
+ * @author Sharptail
+ */
+
+import exceptions.SnomException;
+import tasks.*;
+
+import java.io.IOException;
+import java.nio.file.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    private Path path;
+
+    public Storage(String directory, String filename){
+        this.path = Paths.get(directory, filename);
+        createFolder();
+        createFile();
+    }
+
+    /**
+     * Creates directory with the given path
+     */
+    public void createFolder(){
+        try{
+            Files.createDirectories(path.getParent());
+        }catch(FileAlreadyExistsException e){
+            // Ignored
+        }catch(IOException e){
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Creates file with the given path
+     */
+    public void createFile(){
+        try{
+            Files.createFile(path);
+        }catch(FileAlreadyExistsException e){
+            // Ignored
+        }catch(IOException e){
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Returns an array list of tasks after reading from a save file.
+     *
+     * @return array list of tasks
+     */
+    public ArrayList<Task> readFile() throws SnomException {
+        ArrayList<Task> taskList = new ArrayList<>();
+        try {
+            List<String> lines = Files.readAllLines(path);
+            for(String line: lines){
+                String attr[] = line.split(",");
+                switch(attr[0]){
+                    case "T":
```

Comment:
Should the switch and case have the same indentations?

Reply:
Thank you for spotting my mistake, I must have left out the indentation for this switch case as I were used to another standard. I will modify it soon!